Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -302,7 +302,10 @@ class TimingAnimation extends Animation {
     this.__onEnd = onEnd;
 
     var start = () => {
-      if (this._duration === 0) {
+      // Animations that sometimes have 0 duration and sometimes do not
+      // still need to use the native driver when duration is 0 so as to
+      // not cause intermixed JS and native animations.
+      if (this._duration === 0 && !this._useNativeDriver) {
```

Comment:
There are other ways of solving this, but I'm pretty out-of-my-depth with regards to the RN animation code. Open to suggestions on what the right approach is to solve this issue.

A quicker & dirtier solution is to just not use 0 in TouchableOpacity. A more robust solution might see if it's acceptable to "animate" 0-duration animations in JS even for elements using the native driver, but i'm not sure what the implications would be there.